specification.markdown

```diff
@@ -155,11 +155,40 @@ Class D    | 4      | 0x101FFFFF - 0x1FFFFFFE |   268,338,304
+## Unknown Data Size
+
+An Element Data Size with all VINT_DATA bits set to one is reserved as an indicator that the size of the EBML Element is unknown. The only reserved value for the VINT_DATA of Element Data Size is all bits set to one. An EBML Element with an unknown Element Data Size is referred to as an Unknown-Sized Element. A Master Element MAY be an Unknown-Sized Element; however an EBML Element that is not a Master Element MUST NOT be an Unknown-Sized Element. Master Elements MUST NOT use an unknown size unless the unknownsizeallowed attribute of their EBML Schema is set to true (see [the section on the unknownsizeallowed attribute](#unknownsizeallowed)).
+
+The use of Unknown-Sized Elements allows for an EBML Element to be written and read before the size of the EBML Element is known. Unknown-Sized Element MUST NOT be used or defined unnecessarily; however if the Element Data Size is not known before the Element Data is written, such as in some cases of data streaming, then Unknown-Sized Elements MAY be used. The end of an Unknown-Sized Element is determined by whichever comes first:
+
+- Any EBML Element that is a valid Parent Element of the Unknown-Sized Element according to the EBML Schema, [Global Elements](#global-elements) excluded.
+- Any valid EBML Element according to the EBML Schema, [Global Elements](#global-elements) excluded, that is not a Descendant Element of the Unknown-Sized Element but share a common direct parent, such as a Top-Level Element.
+- Any EBML Element that is a valid Root Element according to the EBML Schema, [Global Elements](#global-elements) excluded.
+- The end of the Parent Element with a known size has been reached.
+- The end of the EBML Document, either when reaching the end of the file or because a new EBML Header started.
+
+Consider an Unknown-Sized Element which EBML path is `\root\level1\level2\elt`. Here are some possible and impossible way to end this element:
+
+EBML Path of next element          | Status
+:----------------------------------|:-----------------------------
+`\root\level1\level2`              | Ends the Unknown-Sized Element, as it is a new Parent Element
+`\root\level1`                     | Ends the Unknown-Sized Element, as it is a new Parent Element
+`\root`                            | Ends the Unknown-Sized Element, as it is a new Root Element
+`\root2`                           | Ends the Unknown-Sized Element, as it is a new Root Element
+`\root\level1\level2\other`        | Ends the Unknown-Sized Element, as they share the same parent
+`\root\level1\level2\elt`          | Ends the Unknown-Sized Element, as they share the same parent
+`\root\level1\level2\elt\inside`   | Doesn't end the Unknown-Sized Element, it's a child of `elt`
+`\root\level1\level2\elt\<global>` | Global Element is valid, it's a child of `elt`
+`\root\level1\level2\<global>`     | Global Element cannot be assumed to have this path, it can only be a child of `elt` while parsing `elt`
```

Comment:
could add

```
\<global>
\root\<global>
\root\level1\<global>
```

Also now we should contextualize the <global> because now we define their path to say `where` they are global. So for instance if the global is defined to not be a child of `elt` but is permitted as a  child of one of elt's ancestors than it could be used to end `elt`.

Reply:
This should go in the "global element" section, not the "unknown size" section. The fact that an element cannot be found at certain level is unrelated to how you can end an unknown length element. Otherwise we're going to need a table for each case where an element is legal or not and how to interpret such ID, etc. This section should only focus on cases where you know an unknown length size element has ended.

Reply:
For example I could add:
> Global Element is valid, it's a child of `elt` (assuming <global> is valid at this path)

But then we would need to do the same for every line above. It seems obvious that it's ends the element because, on top of its position, it's actually a valid element. I'll try add some text before the table to explain that.